pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepositoryUtils.java

```diff
@@ -205,6 +205,9 @@ private static boolean isCommunity(File f) {
     private static void removeMatchingCondition(File element, Function<File, Boolean> condition) {
         if (element.isDirectory()) {
             Stream.of(element.listFiles()).forEach(file -> removeMatchingCondition(file, condition));
+            if (element.list().length == 0) {
```

Comment:
I removed it in: c449ea999c5f1 because it was causing failures at sources generation but unfortunately younger @thescouser89 was dumb and didn't elaborate further. 

It might have been at that time that the folders were not populated properly due to some other PiG issue and the folder was empty and when later on at source generation the sources folder was empty and this deleted it.

I'll assume that you know why you have added this again.

Reply:
this method should not be used in source generation... Do you see it being used somewhere?

Reply:
this method is used to remove community artifact from a product repo. This is to remove empty directories like `junit/junit/4.12/` from the repo zip